solutions/usaco-834.mdx

```diff
@@ -0,0 +1,68 @@
+---
+id: usaco-1037
+source: USACO Bronze 2020 US Open
+title: Cowntact Tracing
+author: Brad Ma
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_sort_silver_open18.html)
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N \cdot T^2)$
+
+<LanguageSection>
+
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class OutOfSorts {
+
+	// hold values
+	static class Entry implements Comparable<Entry> {
+		public int value;
+		public int index;
+
+		public Entry (int v, int i) {
+			value = v;
+			index = i;
+		}
+
+		public int compareTo(Entry other) { // bubble sort
+			if (this.value != other.value) {
+				return this.value - other.value;
+			}
+			return this.index - other.index;
+		}
+	}
+
+	public static void main(String[] args) throws Exception {
+		Kattio io = new Kattio("sort");
+		int n = io.nextInt();
+		Entry[] entries = new Entry[n];
+		for (int i = 0; i < n; i++) {
+			entries[i] = new Entry(io.nextInt(), i);
+		}
+		Arrays.sort(entries); // perform bubble sort
+
+		// values usually only move one spot per iteration
+		int ans = 1;
+		for (int i = 0; i < entries.length; i++) {
+			// count the number of bubbles needed to reach the sorted array
+			// we need to add 1 to account for the final iteration of the algorithm
```

Comment:
not completely sure what these comments mean

Reply:
i can understand why it is confusing but it will make more sense to the person completing the problem